javalib/src/main/scala/java/util/concurrent/atomic/LongAdder.scala

```diff
@@ -1,55 +1,95 @@
-// Ported from Scala.js commit:  6a23632 dated: 1 Jul 2021
-/*
- * Scala.js (https://www.scala-js.org/)
- *
- * Copyright EPFL.
- *
- * Licensed under Apache License 2.0
- * (https://www.apache.org/licenses/LICENSE-2.0).
- *
- * See the NOTICE file distributed with this work for
- * additional information regarding copyright ownership.
- */
+// Ported from JSR 166 revision 1.23
+
 package java.util.concurrent.atomic
 
 import java.io.Serializable
 
-class LongAdder extends Number with Serializable {
-  private[this] var value: Long = 0L
+@SerialVersionUID(7249069246863182397L)
+object LongAdder {
 
-  final def add(x: Long): Unit =
-    value = value + x
+  @SerialVersionUID(7249069246863182397L)
+  private class SerializationProxy(val a: LongAdder) extends Serializable {
```

Comment:
Should `a: LongAdder` be only constructor argument and not a field? Otherwise it looks like a cyclic dependency for me, becouse we would again want to serialize field of LongAdder type in it's serializer. 

```suggestion
  private class SerializationProxy(a: LongAdder) extends Serializable {
```

Also you can point why it's needed (I guess that's it's done to hide the implementation detail of Stripped64 transient fields)